### ISSUE TITLE
Gem render consistency

### ISSUE THREAD

**Issue by nicolas-besnard** [lynndylanhurley/devise_token_auth#87]
Hi,

It appears that the Gem doesn't use the same json render everywhere. Do you think it can be a good idea to generate render with template file such as [jBuilder](https://github.com/rails/jbuilder) or with [rabl](https://github.com/nesquena/rabl) ?

**Comment by nicolas-besnard** [comment]
I'm busy since the beginning of the year. I'll give some of my time on this project when my current company project will be finished.

**Comment by lorenjohnson** [comment]
@nicolas-besnard I would be willing to help-out on this, we need it too. Let me know when you get back to it or if you have notes and would be interested in me getting going on it.

**Comment by nicolas-besnard** [comment]
I'm free this week-end. Should I implement [your solution](https://github.com/lynndylanhurley/devise_token_auth/pull/88#issuecomment-67529816) ? Or do you come with an other alternative ?

**Comment by nicolas-besnard** [comment]
What about a renderer function on the user concern ? Replace [this](https://github.com/lynndylanhurley/devise_token_auth/blob/master/app/controllers/devise_token_auth/sessions_controller.rb#L40-L44) with the following code in the concern :

```ruby
render render json: @resource.create_session_render
```

For example, here's the output of my app on Session#create

```http
{
    "user": {
        "id": 19,
        "first_name": null,
        "last_name": null,
        "email": "besnard.nicolas@gmail.com",
        "created_at": 1424020765,
        "updated_at": 1424020765,
        "birthday": null,
        "gender": null,
        "username": "Clara Rowe",
        "photos": {
            "host": "https://mybucket.s3.amazonaws.com/",
            "profile": null
        }
    }
}
```

**Comment by booleanbetrayal** [comment]
Closing due to inactivity / addressed PR ... can re-open if need be.

**Comment by itkin** [comment]
hello, just found your gem and i'm giving it a try, this render issue bugs me so ended here ... so my point of vue, i guess you have some contraints from client side & oauth2 lib, but : 
- the `success` & `status` booleans seem unusefull since you got this information from the response status (client side the promise resolve or fail consequently)
- don't understand why to return the data sent to the serialized resource to the client, it knows the data it just sent plus you may expose some attributes you don't want to (using simple `as_json`)
- there is still some `as_json` calls in registration controller (which is inconsistent with the custom method set from the user concern)

So to sum up, why not drop all this success / status / data keys in the responses and 
- on success just return the resource using rails (ex-)standard `respond_with` approch. It let the possibility to the user to use his own renderer by adding a view file and it defaults on as_json
- on error just the error list (with the key `:errors`) with no data

If you think it may be interesting i can PR something

**Comment by Animeye** [comment]
I'm also curious what happened with the render cleanup/simplification that was discussed in this issue.  I see in the PR that @nicolas-besnard mentioned making a new PR, but it doesn't look like anything actually happened with that
